tests/integration/csp/assertions/check.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# exit upon any error
+set -o errexit
+
+# exit upon using undeclared variables
+set -o nounset
+
+die(){
+    echo >&2 "$@"
+    exit 1
+}
+
+MODULE_NAME=$(pwd | cut -d'/' -f8)
+
+# get all the assertions (paths to texts files that we want to find in the report)
+# 1 assertion per target, one or some target per modules
+# assertions can be found in the same directory than this file
+declare -a assertions=()
+mapfile -t assertions < <(find . -name "*.txt")
+if [[ ${#assertions[@]} -eq 0 ]]; then 
+    die "Error: No assertions found in module $MODULE_NAME"
+fi
+
+# Since the assertions and the target reports must have the same name 
+# (except their extention name, .out and .txt) we extract the targets names
+# from the assertion array
+# We also use their names to log some informations so keeping an array is
+# somewhat useful
+declare -a targets_name
+for i in "${assertions[@]}"; do
+    filename="$(basename "$i")"
+    targets_name+=("${filename%.*}")
+done
+
+
+# We store the target reports paths in an array with respect to the target_name 
+# array order 
+# That way, our third array is ordered with the 2 other ones and we can safely 
+# use indexes to compare
+declare -a outputs
+for i in "${targets_name[@]}"; do
+    outputs+=("$(find ../../.test/"$MODULE_NAME"/ -name "$i.out")")
+done
+if [[ ${#outputs[@]} -eq 0 ]]; then 
+    die "Error: No targets found in module $MODULE_NAME"
+fi
+
+# A special case is if we don't get the same number of reports as they are targets
+# Wapiti may not detect some targets or it can be due to various misconfigurations
+# So we check the size of each array to be sure they match
+if [[ ${#assertions[@]} -ne ${#targets_name[@]} || ${#targets_name[@]} -ne ${#outputs[@]} ]] ; then
+    die "Error: different number of reports/assertion files, found ${#outputs[@]} outputs for ${#assertions[@]} assertions"
+fi
+
+# Finally, we give grep (in quiet mode) the 2 paths to know weither an assertion's content is 
+# a substring of the report or not 
+for i in "${!outputs[@]}"; do
+    if [[ $( grep -qFf <(cat "${assertions[$i]}") <(cat "${outputs[$i]}")) -ne 0 ]]; then
```

Comment:
A propos du -q : 

>Quiet; do not write anything to standard output.  Exit
              immediately with zero status if any match is found, even
              if an error was detected.

Ça ne posera pas de problème de ne pas remonter les erreurs ?

Reply:
rediriger la sortie standard de grep vers /dev/null te semble être une bonne idée ? Comme ça on garde la sortie d'erreur libre

Reply:
ouaip, ça le ferait bien